py/ringbuf.c

```diff
@@ -142,3 +142,47 @@ int ringbuf_put16(ringbuf_t *r, uint16_t v) {
     ringbuf_put(r, v & 0xff);
     return 0;
 }
+
+// Returns:
+//    1: Success
+//    0: Not enough data available to complete read (try again later)
+//   -1: Requested read is larger than buffer - will never succeed
+int ringbuf_read(ringbuf_t *r, void *data, size_t data_len) {
```

Comment:
This and `ringbuf_write()` are very similar to `ringbuf_get_n()` and `ringbuf_put_n()`, with some additional return values. Did they come from upstream or did you add them?

I'd rather they take `uint8_t*` instead of `void *`. Do the casting on the calling side; that ensures the user knows what they take.

These could wrap the `get` and `put` routines, and could be `get_all` and `put_all` or something like that.

Reply:
Modified `espnow` to use existing `ringbuf` functions.